### PR TITLE
indents(php): fix indents in return statements and paran expressions

### LINKED ISSUE #4848: php indentation is broken when using 'return new class'
### Describe the bug

indentation breaks in php when writing `return new class extends OtherClass` instead of `class ClassName extends OtherClass`

not sure if related to https://github.com/nvim-treesitter/nvim-treesitter/issues/2497

### To Reproduce

1. create a file with contents:

```php
<?php
return new class extends Migration
{
    public function up()
    {
        Schema::create('password_resets', function (Blueprint $table) {
            $table->string('email')->primary();
            $table->string('token');
            $table->timestamp('created_at')->nullable();
        });
    }
}
```

2. go to line 5 and/or 6 and press `o` - observe that cursor is *not* where it should be if tree-sitter indent is functioning properly.

## expected behavior

4. change the second line of the php file to `class Users extends Migration`
5. go to line 5 and/or 6 and press `o` - observe that the cursor *is* where it should be, indent is working.

## video of reproduction

https://github.com/nvim-treesitter/nvim-treesitter/assets/83531587/4aec9d07-bbdd-41da-b3fa-174090f4dff9

### Output of `:checkhealth nvim-treesitter`

```markdown
==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.8 (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v20.2.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: cc (GCC) 13.1.1 20230429
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "6.3.2-zen1-1-zen",
  sysname = "Linux",
  version = "#1 ZEN SMP PREEMPT_DYNAMIC Thu, 11 May 2023 16:40:19 +0000"
} ~

Parser/Features         H L F I J
  - css                 ✓ . ✓ ✓ ✓
  - dart                ✓ ✓ ✓ ✓ ✓
  - fennel              ✓ ✓ ✓ . ✓
  - go                  ✓ ✓ ✓ ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - php                 ✓ ✓ ✓ ✓ ✓
  - python              ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~
```

### Output of `nvim --version`

```text
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by calebdw**
Hello!

This PR fixes several double indents with return statements. It also correctly indents parenthesized_expressions.

Note: I wasn't sure if this should target `main` or `master`---I would like this to be immediately usable after it is merged if that helps determine which one to use.

Thanks!

Closes #4848.

### CODE REVIEW

File: runtime/queries/php_only/indents.scm [hunk 1]
```diff
@@ -12,6 +13,12 @@
   (case_statement)
 ] @indent.begin
 
+(return_statement [
+  (object_creation_expression)
+  (anonymous_function_creation_expression)
+  (match_expression)
+]) @indent.dedent
```
Comment (lucario387) [inline runtime/queries/php_only/indents.scm:20]: You mean dedent for the inner expression, or the outer?

Comment (calebdw) [reply runtime/queries/php_only/indents.scm:20]: The outer (return statement)---the inner expression is properly indented

**Comment by lucario387** [comment]
This should target `master`.

Additionally, remember to run `./scripts/format-queries.lua queries/php`

File: tests/indent/php/issue-4848.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+return function () {
+};
+
+return match (
+) {
+};
+
+return match () {
+        // double indent
```
Comment (lucario387) [inline tests/indent/php/issue-4848.php:32]: I don't think it's intended for this to have double indent :thinking:

Comment (calebdw) [reply tests/indent/php/issue-4848.php]: Good catch, I forgot to remove this---it was double indented before the fix

**Comment by calebdw** [comment]
I also fixed indentation issues with chained method calls and closures inside of them. Most cases are working, but there's one case that I just can't figure out:

```php
$this->each(function (
    $a,
) {
    return $a;
});

// gets formatted as

$this->each(function (
    $a,
) {
        return $a;
    });
```

**Comment by lucario387** [comment]
I will look at it later (as in tonight or tomorrow)

**Review by lucario387 — APPROVED** [review]

**Comment by amaanq** [comment]
thanks

**Pull request merged**